### PR TITLE
Array of files support for FromFileSchemaProvider

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    |  ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #58 

New `FromFilesSchemaProvder` can use so:

```php
FromFilesSchemaProvider::class => [
	'files' => [
	  'module1/cycle-schema.php',
	  'module2/cycle-schema.php',
	  'module3/cycle-schema.php',
	],
],
```

### CODE REVIEW

File: tests/Schema/Provider/FromFilesSchemaProvider/FromFilesSchemaProviderTest.php [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+    public function testRead(): void
+    {
+        $schemaProvider = $this->createSchemaProvider();
+
+        $data = $schemaProvider
+            ->withConfig([
+                'files' => [
+                    '@dir/schema1.php',
+                    '@dir/schema-not-exists.php', // not exists files should be silent
```
Comment (roxblnfk) [inline tests/Schema/Provider/FromFilesSchemaProvider/FromFilesSchemaProviderTest.php:83]: Why not throw an exception if the file is missing?

File: src/Schema/Provider/FromFilesSchemaProvider.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            if (is_file($file)) {
+                foreach (require $file as $role => $definition) {
+                    if (array_key_exists($role, $schema)) {
+                        throw new LogicException('The "' . $role . '" role already exists in the DB schema.');
+                    }
+                    $schema[$role] = $definition;
+                }
+            }
+        }
+        return empty($schema) ? null : $schema;
```
Comment (roxblnfk) [inline src/Schema/Provider/FromFilesSchemaProvider.php:60]: Should we convert an empty array to the NULL if empty arrays are intentionally specified in the merged files?

**Review by roxblnfk — COMMENTED** [review]
Not bad 👍  I'll look at the tests a little later.

File: tests/Schema/Provider/FromFilesSchemaProvider/FromFilesSchemaProviderTest.php (hunk 1 above)
Comment (vjik) [reply tests/Schema/Provider/FromFilesSchemaProvider/FromFilesSchemaProviderTest.php:83]: According to the logic described [here](https://github.com/yiisoft/yii-cycle/blob/master/docs/en/reading-schema.md) there should be no exception.

File: src/Schema/Provider/FromFilesSchemaProvider.php (hunk 2 above)
Comment (vjik) [reply src/Schema/Provider/FromFilesSchemaProvider.php:60]: Difficult question.

I see two variants:

1) If read ALL files, then return schema, else `null`.
2) If read AT LEAST ONE file, then return schema, else `null`.

File: tests/Schema/Provider/FromFilesSchemaProvider/FromFilesSchemaProviderTest.php (hunk 1 above)
Comment (vjik) [reply tests/Schema/Provider/FromFilesSchemaProvider/FromFilesSchemaProviderTest.php:134]: Add `strict` parameter.

File: src/Schema/Provider/FromFilesSchemaProvider.php (hunk 2 above)
Comment (vjik) [reply src/Schema/Provider/FromFilesSchemaProvider.php]: Add `strict` parameter.

**Review by roxblnfk — APPROVED** [review]

**Review by rustamwin — COMMENTED** [review]
How about wildcard support?

**Comment by vjik** [comment]
> How about wildcard support?

This is good idea. But I think for next PR :)

**Comment by roxblnfk** [comment]
> How about wildcard support?

> This is good idea. But I think for next PR :)

Everything is correct.
I was planning to create an Issue about this after merging current PR

**Review by rustamwin — APPROVED** [review]
Alright.

**Pull request merged**